datadog/cmd/main/main.go

```diff
@@ -213,6 +214,90 @@ func (d *DatadogCostSource) getDDCostsForWindow(window opencost.Window, listPric
 	// datadog's usage API sometimes provides usages that get counted multiple times
 	// this post processing stage de-duplicates those usages and costs
 	postProcess(&ccResp)
+
+	// query from the first of the window's month until the window end's day so that we can properly adjust for the
+	// cumulative nature of the response
+	startDate := time.Date(window.Start().Year(), window.Start().Month(), 1, 0, 0, 0, 0, time.UTC)
+	endDate := time.Date(window.End().Year(), window.End().Month(), window.End().Day(), 0, 0, 0, 0, time.UTC)
```

Comment:
I think we might want this, just in case:
```suggestion
	// query from the first of the window's month until the window end's day so that we can properly adjust for the
	// cumulative nature of the response
	startDate := time.Date(window.Start().UTC().Year(), window.Start().UTC().Month(), 1, 0, 0, 0, 0, time.UTC)
	endDate := time.Date(window.End().UTC().Year(), window.End().UTC().Month(), window.End().UTC().Day(), 0, 0, 0, 0, time.UTC)
```

I'm open to alternatives

Reply:
I'm closed to alternatives in this case, I like your solution.